ndio/remote/neurodata.py

```diff
@@ -1189,31 +1214,164 @@ def create_channel(self, token, name, channel_type, dtype, readonly):
             return True
 
     @_check_token
-    def delete_channel(self, token, name):
+    def create_channels(self, dataset, token, new_channels_data):
+
         """
-        Delete an existing channel on the Remote. Be careful!
+        Creates channels given a dictionary in 'new_channels_data'
+        , 'dataset' name, and 'token' (project) name.
 
         Arguments:
-            token (str): The token the new channel should be deleted from
-            name (str): The name of the channel to delete
+            token (str): Token to identify project
+            dataset (str): Dataset name to identify dataset to download from
+            new_channels_data (dict): New channel data to upload into new channels
 
         Returns:
-            bool: True if successful, False otherwise.
+            bool: Process completed succesfully or not
+        """
+
+        channels = {}
+        for channel_new in new_channels_data:
+
+            self._check_channel(channel_new.name)
+
+            if channel_new.channel_type not in ['image', 'annotation']:
+                raise ValueError('Channel type must be ' +
+                                 'neurodata.IMAGE or neurodata.ANNOTATION.')
+
+            if channel_new.readonly * 1 not in [0, 1]:
+                raise ValueError("readonly must be 0 (False) or 1 (True).")
+
+            channels[channel_new.name] = {
+                "channel_name": channel_new.name,
+                "channel_type": channel_new.channel_type,
+                "datatype": channel_new.dtype,
+                "readonly": channel_new.readonly * 1
+            }
+        req = requests.post(self.url("/{}/project/".format(dataset) + "{}".format(token)), json={"channels" : {channels}})
+
+        if req.status_code is not 201:
+            raise RemoteDataUploadError('Could not upload {}'.format(req.text))
+        else:
+            return True
+
+    @_check_token
+    def delete_channel(self, token, channel_names):
+        """
+
+        Deletes channels listed in 'channel_names' and in the
+        project folder 'token'.
+
+        Arugments:
+            token (str): Token to identify project
+            channel_names (list): List containing names of channels to delete
+
+        Returns:
+            bool: Process completed succesfully or not
 
-        Raises:
-            RemoteDataUploadError: If the upload fails for some reason.
         """
-        req = requests.post(self.url("{}/deleteChannel/".format(token)), json={
-            "channels": [name]
+        url = self.url()[:-4] + "/ca/{}/deleteChannel/".format(token)
+        print url
+        req = requests.post(url, json={
+            "channels": [channel_names]
         })
 
         if req.status_code is not 200:
+            print 'error is',str(req.status_code), '\n'
```

Comment:
This should be in the error message and not a printout. In general we raise errors over printing and errors because its better catchable and clearer that something actually went wrong.